lib/rucio/daemons/tracer/kronos.py

```diff
@@ -496,6 +492,11 @@ def run(once=False, threads=1, sleep_time_datasets=60, sleep_time_files=60):
     if rucio.db.sqla.util.is_old_db():
         raise DatabaseException('Database was not updated, daemon won\'t start')
 
+    try:
+        dataset_wait = config_get_int('tracer-kronos', 'dataset_wait')
```

Comment:
I don't think dataset_wait is intended to be used as sleep time.

Reply:
To me, it looks like `dataset_wait` is a remnant from before we had sleep times. I can’t see a reason why it’s needed at all. Refer to commit https://github.com/rucio/rucio/commit/d26d4239bc04c92f7a06e6d1b27c65778b9961fa.

Reply:
For me it's a mechanism to avoid running kronos.dataset all together during a small time window after the start of the daemon. Probably to avoid issues during releases because of heartbeat rebalancing. It's more an equivalent of `partition_wait_time` than of `sleep`

Reply:
I'll leave it as it used to be for now. But it does seem a bit redundant since the `run_daemon` method inserts the `sleep_time` between the subsequent runs of the `run_once` method. If we want to control the `partition_wait_time`, maybe we should rename the config parameter in rucio.cfg as partition_wait_time or explicitly set `patiotion_wait_time=dataset_wait` at some place.

Reply:
I find myself disagreeing with this course of action.

The original intent of `dataset_wait` should be clear. I fail to see an argument for keeping it any longer. `run_daemons()`’s `partition_wait_time` and `sleep_time` can replace it, with the former having a hard-coded value (as we do everywhere else) and the latter being set through `--sleep-time-datasets`.